### DISCUSSION TITLE
I am getting error while deploying on zksync.

### DISCUSSION THREAD

**Discussion by imranshafiq55** [Cyfrin/foundry-full-course-cu#2915] [Q&A]
This is the error that I am getting:
Error: "underlying network changed (event=\"changed\", network={\"chainId\":300,\"name\":\"zkSync Sepolia Testnet\"}, detectedNetwork={\"name\":\"homestead\",\"chainId\":1,\"ensAddress\":\"0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e\"}, code=NETWORK_ERROR, version=providers/5.7.2)"

![Screenshot (93)](https://github.com/user-attachments/assets/3b17faef-d1ed-461d-9831-325774726e8d)

**Comment by EngrPips** [comment]
Hello @imranshafiq55, Are you sure your `Metamask` is on `zksync sepolia` chain?

**Reply by imranshafiq55** [reply]
![Screenshot (94)](https://github.com/user-attachments/assets/485f6495-542e-47c0-b5c3-beb04f7c3161)
Yes check this.
![Screenshot (95)](https://github.com/user-attachments/assets/011a4eca-396d-4684-a6b1-a2f9d11968e9)

**Reply by EngrPips** [reply]
Then,  it should work as expected. I just deployed a contract to `zksync sepolia` without any issue just now 

![Screenshot 2024-11-26 at 11 51 27](https://github.com/user-attachments/assets/d8ba99cd-1462-4acf-a899-264cc21379c1)

![Screenshot 2024-11-26 at 11 54 01](https://github.com/user-attachments/assets/797e0893-b156-47c3-8c98-004fa7d0ae68)

![Screenshot 2024-11-26 at 11 54 29](https://github.com/user-attachments/assets/9f0c0783-ac6b-481f-939d-573816a16c30)

**Reply by imranshafiq55** [reply]
ok let me check again

**Reply by EngrPips** [reply]
It works seamlessly for me.

**Comment by imranshafiq55** [comment]
again I am getting same issue.

**Comment by imranshafiq55** [comment]
![image](https://github.com/user-attachments/assets/c8b192db-c1c2-4f50-a231-dce3669ba884)

**Reply by EngrPips** [reply]
Please Uncheck the `verify contract` box and then retry.

**Reply by imranshafiq55** [reply]
again same issue persists.

**Reply by imranshafiq55** [reply]
![image](https://github.com/user-attachments/assets/f81dee99-8966-4b03-a4e8-f326e7c7bc56)

**Reply by EngrPips** [reply]
Do you have the file inside of a `contracts` folder? Like I have it in the image below.

![Screenshot 2024-11-26 at 12 03 39](https://github.com/user-attachments/assets/be0b571e-0f57-48f0-ad82-78c10a855e7a)

**Reply by imranshafiq55** [reply]
I have done this too but again same issue
![image](https://github.com/user-attachments/assets/1f599a60-5dec-4d80-8c2d-f0bbf2d135e6)

**Reply by EngrPips** [reply]
Tough. It seems like a problem from your end because it works fine for me.

**Reply by imranshafiq55** [reply]
let me check this. Thank you for your help and support.

**Reply by EngrPips** [reply]
No worries at all, I hope you figure it out.
